src/routes/likernft/book/purchase.ts

```diff
@@ -532,6 +533,38 @@ router.get(
   },
 );
 
+router.get('/free', jwtOptionalAuth('read:nftbook'), async (req, res, next) => {
+  try {
+    const freeBooks = await getFreeBooksForUser(req.user?.evmWallet);
+    res.json(freeBooks);
+  } catch (err) {
+    next(err);
+  }
+});
+
+router.post('/free', jwtAuth('write:nftbook'), async (req, res, next) => {
+  try {
+    const { user } = req;
+    if (!user) throw new ValidationError('UNAUTHORIZED', 401);
+    const { class_id: classId } = req.body;
```

Comment:
For post data, no need to use snakecase?
```suggestion
    const { classId } = req.body;
```